### DISCUSSION TITLE
`x-on:*` should await when given an async function

### DISCUSSION THREAD

**Discussion by jherdman** [alpinejs/alpine#2902] [3. Feature ideas]
I ran into a problem recently when integrating with the wondering [Analytics](https://www.npmjs.com/package/analytics) lib.

I want to make a call to my analytics service to identify a user on submit of a form. I want to ensure this event finishes. Fortunately `analytics.identify` is an async.

Given this form:

```html
<form @submit="analytics.identify({ email: userEmail })">
```

I want the submit hook (or any x-on event) to wait for the async function I've provided to return.

**Comment by SimoTod** [comment]
https://alpinejs.dev/advanced/async
Have you tried using await?

**Reply by jherdman** [reply]
I have not! Thank you for pointing this out. Kinda bummed I won't get to write a PR though! Next time.
